src/apps/answers/service.py

```diff
@@ -1922,9 +1923,38 @@ async def is_flow_finished(self, submit_id: uuid.UUID, answer_id: uuid.UUID) ->
         return self._is_activity_last_in_flow(applet_full, activity_id, flow_id)
 
     async def create_report(
-        self, submit_id: uuid.UUID, answer_id: uuid.UUID | None = None
+        self,
+        submit_id: uuid.UUID,
+        answer_id: uuid.UUID | None = None,
+        activity_id: uuid.UUID | None = None,
+        report_name: str | None = None,
     ) -> ReportServerResponse | None:
         answers = await AnswersCRUD(self.answers_session).get_by_submit_id(submit_id, answer_id)
+        subscale_table_data: list[SubScaleLookupTable] | None = None
+        subscale_name = ""
+        scoring_type = ""
+
+        if activity_id:
+            activity_data = await ActivitiesCRUD(self.session).get_by_id(activity_id)
+            reports = activity_data.scores_and_reports.get("reports", [])
+            if not reports:
+                return None  # Early return if no reports are found
+
+            selected_report = next((report for report in reports if report["name"] == report_name), reports[0])
+
+            scoring_type = selected_report.get("scoring_type", "")
+            if scoring_type == "score":
```

Comment:
Just omit the second parameter for .get() method, it will return None when field is missing

Use enum ScoringType in this IF